sap-utils/src/main/java/org/wso2/ei/module/sap/consumer/BapiHandlerFactory.java

```diff
@@ -16,55 +16,78 @@
  * under the License.
  */
 
-package org.ballerinalang.sap.nativeimpl.consumer;
+package org.wso2.ei.module.sap.consumer;
 
 import com.sap.conn.jco.JCoFunction;
 import com.sap.conn.jco.server.JCoServerContext;
 import com.sap.conn.jco.server.JCoServerFunctionHandler;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-import org.ballerinalang.bre.Context;
-import org.ballerinalang.bre.bvm.CallableUnitCallback;
-import org.ballerinalang.connector.api.Resource;
+import org.ballerinalang.jvm.BRuntime;
+import org.ballerinalang.jvm.types.AttachedFunction;
+import org.ballerinalang.jvm.values.ErrorValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.jvm.values.connector.CallableUnitCallback;
+import org.ballerinalang.jvm.values.connector.Executor;
 import org.ballerinalang.model.values.BError;
-import org.ballerinalang.sap.utils.SapUtils;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.ei.module.sap.utils.SapUtils;
 
 import java.util.Map;
 
+import static org.wso2.ei.module.sap.utils.SapConstants.RESOURCE_ON_MESSAGE;
+
 /**
  * This class handles the BAPI calls that are returned from the SAP gateway.
  * <p>
  * This class encapsulates each BAPI/RFC request received.
  * by handling  and converting them to Strings.
  * </p>
  */
-public class RFCConnectionHandler implements JCoServerFunctionHandler {
+public class BapiHandlerFactory implements JCoServerFunctionHandler {
+
+    private static Logger log = LoggerFactory.getLogger("ballerina");
 
-    private static final Log log = LogFactory.getLog(RFCConnectionHandler.class);
     private ResponseCallback callback;
-    private Map<String, Resource> sapResource;
-    private Context context;
+    private Map<String, AttachedFunction> sapResource;
+    private final BRuntime runtime;
+//    private Context context;
+
+    BapiHandlerFactory(Map<String, AttachedFunction> sapService) {
 
-    RFCConnectionHandler(Map<String, Resource> sapService, Context context) {
         this.sapResource = sapService;
         callback = new ResponseCallback();
-        this.context = context;
+//        this.context = context;
```

Comment:
Can we remove this comment if not required?

Reply:
Fixed. Thanks.